lib/surface/compiler/parser.ex

```diff
@@ -206,43 +218,54 @@ defmodule Surface.Compiler.Parser do
     [buffer | buffers]
   end
 
-  defp transtate_attrs(attrs) do
-    Enum.map(attrs, &translate_attr/1)
+  defp transtate_attrs([]), do: {:ok, []}
+
+  defp transtate_attrs([attr | attrs]) do
+    with {:ok, translated_attr} <- translate_attr(attr),
+         {:ok, translated_attrs} <- transtate_attrs(attrs) do
+      {:ok, [translated_attr | translated_attrs]}
+    end
   end
 
-  defp translate_attr({name, {:string, value, %{delimiter: ?"}}, %{line: line}}) do
-    {name, value, %{line: line}}
+  defp translate_attr({name, {:string, value, %{delimiter: ?"}}, meta}) do
+    {:ok, {name, value, to_meta(meta)}}
   end
 
-  defp translate_attr({name, {:string, "true", %{delimiter: nil}}, %{line: line}}) do
-    {name, true, %{line: line}}
+  defp translate_attr({name, {:string, "true", %{delimiter: nil}}, meta}) do
+    {:ok, {name, true, to_meta(meta)}}
   end
 
-  defp translate_attr({name, {:string, "false", %{delimiter: nil}}, %{line: line}}) do
-    {name, false, %{line: line}}
+  defp translate_attr({name, {:string, "false", %{delimiter: nil}}, meta}) do
+    {:ok, {name, false, to_meta(meta)}}
   end
 
-  defp translate_attr({name, {:string, value, %{delimiter: nil}}, %{line: line}}) do
+  defp translate_attr({name, {:string, value, %{delimiter: nil}}, meta}) do
     case Integer.parse(value) do
       {int_value, ""} ->
-        {name, int_value, %{line: line}}
+        {:ok, {name, int_value, to_meta(meta)}}
 
       _ ->
-        message = "unexpected value for attribute \"#{name}\""
-        raise %ParseError{line: line, column: 1, message: message}
+        {:error,
+         %ParseError{
+           line: meta.line,
+           column: meta.column,
+           file: meta.file,
+           message: "unexpected value for attribute \"#{name}\""
+         }}
```

Comment:
Same here. This makes the callers more polluted in order to handle the return value as well as the code in each `translate_attr` variant. I'd just leave it as an exception.